numpy/lib/_function_base_impl.py

```diff
@@ -506,6 +521,10 @@ def average(a, axis=None, weights=None, returned=False, *,
     """
     a = np.asanyarray(a)
 
+    # cast axis to tuple if not already
+    if axis is not None and np.array(axis).ndim == 0:
```

Comment:
No need to transform into an array. Elsewhere we just do `if axis is not None and not isinstance(axis, tuple)`. Though, for the below it is helpful if `axis` is normalized to be between 0 and ndim, and that can handily be done at the same time as making an `int` a tuple, using:
```
if axis is not None:
    axis = _nx.normalize_axis_tuple(axis, a.ndim, argname="axis")
```